instrumentation/aws-lambda-1.0/library/src/main/java/io/opentelemetry/instrumentation/awslambda/v1_0/AwsLambdaTracer.java

```diff
@@ -19,12 +21,30 @@
 import io.opentelemetry.api.trace.attributes.SemanticAttributes.FaasTriggerValues;
 import io.opentelemetry.context.Scope;
 import io.opentelemetry.instrumentation.api.tracer.BaseTracer;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
 import java.util.Collections;
 import java.util.Map;
+import org.checkerframework.checker.nullness.qual.Nullable;
 
 public class AwsLambdaTracer extends BaseTracer {
 
   private static final String AWS_TRACE_HEADER_ENV_KEY = "_X_AMZN_TRACE_ID";
+  private static final MethodHandle GET_FUNCTION_ARN;
+
+  static {
+    MethodHandles.Lookup lookup = MethodHandles.publicLookup();
```

Comment:
I wonder if we can just set a boolean by checking with normal reflection and guard the call to the function, given Java doesn't link until call-time? I'm not aware of any advantage of method handles for guarding against different versions vs that (code is a bit simpler IMO) but maybe no one has ever told me them yet :P

Reply:
Yeah, we certainly can -- I used `MethodHandle` without thinking because that's kind of a default in the agent 😅

Reply:
Hmm, there's one issue that appears when I use the method directly: I need to bump the AWS lambda API to 1.2.0, which also requires bumping javaagent lib version to 1.2.0 because muzzle'll fail. That pretty much means dropping support for <1.2.0... @anuraaga are we OK with that?

Reply:
Ah I guess muzzle is too smart for enough - in that case let's stick to the current approach!